src/main/java/org/reaktivity/nukleus/socks/internal/stream/client/AcceptStreamProcessor.java

```diff
@@ -0,0 +1,444 @@
+
+    private void handleConnectThrottleBeforeHandshake(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        switch (msgTypeId)
+        {
+        case WindowFW.TYPE_ID:
+            final WindowFW window = context.windowRO.wrap(buffer, index, index + length);
+            connectWindowBytes += window.update();
+            connectWindowFrames += window.frames();
+            correlation.nextAcceptSignal().accept(isConnectReplyStateReady);
+            break;
+        case ResetFW.TYPE_ID:
+            final ResetFW reset = context.resetRO.wrap(buffer, index, index + length);
+            handleReset(reset);
+            break;
+        default:
+            // ignore
+            break;
+        }
+    }
+
+    private void handleWindow(
+        WindowFW window)
+    {
+        final int targetWindowBytesDelta = window.update();
+        final int targetWindowFramesDelta = window.frames();
+
+        final int sourceWindowBytesDelta = targetWindowBytesDelta + sourceWindowBytesAdjustment;
+        final int sourceWindowFramesDelta = targetWindowFramesDelta + sourceWindowFramesAdjustment;
+
+        sourceWindowBytesAdjustment = Math.min(sourceWindowBytesDelta, 0);
+        sourceWindowFramesAdjustment = Math.min(sourceWindowFramesDelta, 0);
+
+        if (sourceWindowBytesDelta > 0 || sourceWindowFramesDelta > 0)
+        {
+            doWindow(correlation.acceptThrottle(), correlation.acceptStreamId(), Math.max(sourceWindowBytesDelta, 0),
+                Math.max(sourceWindowFramesDelta, 0));
+        }
+    }
+
+    private void handleReset(
+        ResetFW reset)
+    {
+        doReset(correlation.acceptThrottle(), correlation.acceptStreamId());
+    }
+
+    @Signal
+    public void attemptNegotiationRequest(boolean isConnectReplyBeginFrameReceived)
+    {
+        if (dataNegotiationRequestFW == null)
+        {
+            if (NO_SLOT == (slotIndex = context.bufferPool.acquire(correlation.connectStreamId())))
+            {
+                doReset(correlation.acceptThrottle(), correlation.acceptStreamId());
+                return;
+            }
+            MutableDirectBuffer acceptBuffer = context.bufferPool.buffer(slotIndex);
+            SocksNegotiationRequestFW socksNegotiationRequestFW = context.socksNegotiationRequestRW
+                .wrap(acceptBuffer, DataFW.FIELD_OFFSET_PAYLOAD, acceptBuffer.capacity())
+                .version((byte) 0x05)
+                .nmethods((byte) 0x01)
+                .method(new byte[]{0x00})
+                .build();
+
+            dataNegotiationRequestFW = context.dataRW.wrap(acceptBuffer, 0, acceptBuffer.capacity())
+                .streamId(correlation.connectStreamId())
+                .payload(p -> p.set(
+                    socksNegotiationRequestFW.buffer(),
+                    socksNegotiationRequestFW.offset(),
+                    socksNegotiationRequestFW.sizeof()))
+                .extension(e -> e.reset())
```

Comment:
you should probably propagate the authorization field though... here and everywhere